src/test/java/com/contrastsecurity/maven/plugin/it/stub/ContrastAPIStubExtension.java

```diff
@@ -0,0 +1,102 @@
+package com.contrastsecurity.maven.plugin.it.stub;
+
+import java.util.Optional;
+import org.junit.jupiter.api.extension.AfterEachCallback;
+import org.junit.jupiter.api.extension.BeforeEachCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.ExtensionContext.Namespace;
+import org.junit.jupiter.api.extension.ParameterContext;
+import org.junit.jupiter.api.extension.ParameterResolutionException;
+import org.junit.jupiter.api.extension.ParameterResolver;
+
+/**
+ * JUnit 5 test extension that provides test authors with a stubbed instance of the Contrast API.
+ */
+public final class ContrastAPIStubExtension
+    implements ParameterResolver, BeforeEachCallback, AfterEachCallback {
+
+  /**
+   * Starts the {@link ContrastAPI}
+   *
+   * @param context JUnit context
+   */
+  @Override
+  public void beforeEach(final ExtensionContext context) {
+    final ContrastAPI contrast = getContrastAPI(context);
+    contrast.start();
+  }
```

Comment:
When I starting writing the `ContrastAPIStub`, I wasn't sure if it was going to be mutable or not. For example, I thought maybe it would expose some methods that each test would use to perform some configuration like `stubbedDownloadFails()`. Ultimately, I didn't need this yet, so yes the stub is this immutable thing that all the tests can share.

I'll change it to a `beforeAll()` / `afterAll()` (the annotation will move to the test class instead of the test method) and that should get the performance benefits you're looking for. If we ultimately decide that the server stub should be mutable, it's easy enough to change it back.